File: src/strings/strings.zig
const std = @import("std");

test "concat" {
    const hello = "Hello";
    const world = ", world!";
    try std.testing.expectEqualStrings("Hello, world!", hello ++ world);
}

test "runtime concat" {
    const hello = "Hello";
    const world = ", world!";
    const hello_world = try std.mem.concat(std.testing.allocator, u8, &.{ hello, world });
    try std.testing.expectEqualStrings("Hello, world!", hello_world);
}

test "contains" {
    const hello = "Hello, world!";
    try std.testing.expect(std.mem.containsAtLeast(u8, hello, 1, "Hello"));
    try std.testing.expect(!std.mem.containsAtLeast(u8, hello, 1, "Goodbye"));
}

File: src/strings/strings.cs
using System.Text;
using Xunit;

namespace ZigForDotnetDevs;

public class Strings
{
    [Fact]
    public void Concat()
    {
        var hello = "Hello";
        var world = ", world!";
        Assert.Equal("Hello, world!", hello + world);
    }

    [Fact]
    public void ConcatUsingStringBuilder()
    {
        var stringBuilder = new StringBuilder();
        stringBuilder.Append("Hello");
        stringBuilder.Append(", world!");
        Assert.Equal("Hello, world!", stringBuilder.ToString());
    }

    [Fact]
    public void Contains()
    {
        var hello = "Hello, world!";
        Assert.True(hello.Contains("Hello"));
        Assert.False(hello.Contains("Goodbye"));
    }
}
